centreon/src/Core/HostCategory/Application/UseCase/DeleteHostCategory/DeleteHostCategory.php

```diff
@@ -0,0 +1,89 @@
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\NoContentResponse;
+use Core\Application\Common\UseCase\NotFoundResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\Repository\WriteHostCategoryRepositoryInterface;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+
+final class DeleteHostCategory
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private WriteHostCategoryRepositoryInterface $writeHostCategoryRepository,
+        private ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private ReadAccessGroupRepositoryInterface $readAccessGroupRepositoryInterface,
+        private ContactInterface $user
+    ) {
+    }
+
+    public function __invoke(int $hostCategoryId, PresenterInterface $presenter): void
+    {
+        try {
+            if ($this->user->isAdmin()) {
+                if ($this->readHostCategoryRepository->exists($hostCategoryId)) {
+                    $this->writeHostCategoryRepository->deleteById($hostCategoryId);
+                    $presenter->setResponseStatus(new NoContentResponse());
+                } else {
+                    $this->error('Host category not found', [
+                        'hostcategory_id' => $hostCategoryId,
+                    ]);
+                    $presenter->setResponseStatus(new NotFoundResponse('Host category'));
+                }
+            } elseif ($this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)) {
+                $accessGroups = $this->readAccessGroupRepositoryInterface->findByContact($this->user);
+
+                if ($this->readHostCategoryRepository->existsByAccessGroups($hostCategoryId, $accessGroups)) {
+                    $this->writeHostCategoryRepository->deleteById($hostCategoryId);
+                    $presenter->setResponseStatus(new NoContentResponse());
+                } else {
+                    $this->error('Host category not found', [
+                        'hostcategory_id' => $hostCategoryId,
+                        'accessgroups' => $accessGroups
+                    ]);
+                    $presenter->setResponseStatus(new NotFoundResponse('Host category'));
+                }
+            } else {
+                $this->error('User doesn\'t have sufficient rights to see host category', [
+                    'user_id' => $this->user->getId(),
+                ]);
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse('You are not allowed to delete host categories')
+                );
+            }
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(new ErrorResponse('Error while deleting host category'));
```

Comment:
The message needs to be in a `HostCategoryException` class.
⚠️ update the test.